R/cfr.r

```diff
@@ -0,0 +1,232 @@
+                            }
+real variant_nb_rng(int y, real mu, real phi, real alpha,
+                    real f, int cases, int effect) {
+    real scaled_cases;
+    if (effect) {
+      scaled_cases = (mu + alpha * f) * cases;
+    }else {
+      scaled_cases = (1 + (alpha - 1) * f) * mu * cases;
+    }
+    return  neg_binomial_2_rng(scaled_cases, phi);
+                            }
+"
+  stanvars <- c(stanvar(block = "functions", scode = stan_funs),
+                stanvar(block = "data",
+                        scode = "  real f[N];",
+                        x = data$prop_sgtf,
+                        name = "f"),
+                stanvar(block = "data",
+                        scode = "  int cases[N];",
+                        x = data$cases,
+                        name = "cases"),
+                stanvar(block = "data",
+                        scode = "  int effect[1];",
+                        x = array(as.numeric(additive)),
+                        name = "effect")
+  )
+  return(stanvars)
+}
+
+# define model function
+nb_model <- function(form, iter = 2000, data = deaths_with_cov,
+                     additive = FALSE, ...) {
+  # define priors
+  if (additive) {
+    priors <- c(prior(normal(0, 0.01), class = alpha))
+  }else{
+    priors <- c(prior(lognormal(0, 0.5), class = alpha))
+  }
+  message("Fitting: ", as.character(form))
+  brm(formula = form,
+      family = variant_nb(additive = additive),
+      prior = priors,
+      data,
+      stanvars = make_stanvars(data, additive = additive),
+      control = list(adapt_delta = 0.99),
+      warmup = 1000, iter = iter, ...)
+}
+# Fit models --------------------------------------------------------------
+models <- list()
+# define models to fit
+models[["intercept"]] <- as.formula(deaths ~ 1)
+models[["cases"]] <- as.formula(deaths ~ s(normalised_cases, k = 5))
+models[["region"]] <- as.formula(deaths ~ region)
+models[["time"]] <- as.formula(deaths ~ s(time, k = 9))
+models[["utla"]] <- as.formula(deaths ~ (1 | utla))
+models[["all"]] <- as.formula(deaths ~ s(normalised_cases, k = 5) + region + (1 | utla))
+models[["all_with_residuals"]] <- as.formula(deaths ~ s(normalised_cases, k = 5) + s(time, k = 5) + region + (1 | utla))
+models[["all_with_regional_residuals"]] <- as.formula(deaths ~ s(normalised_cases, k = 5) + s(time, k = 5, by = region) + (1 | utla))
+
+# core usage
+if (no_cores <= 4) { 
+  options(mc.cores = no_cores)
+  mc_cores <- 1
+}else{
+  options(mc.cores = 4)
+  mc_cores <- ceiling(no_cores / 4) 
+}
+# fit models
+warning("Fitting models sequentially due to mclapply issues")
+fits <- list()
+fits[["multiplicative"]] <- lapply(models, nb_model)
+fits[["additive"]] <- lapply(models, nb_model, additive = TRUE)
+
+# variant effect ----------------------------------------------------------
+extract_variant_effect <- function(x, additive = FALSE) {  
+  samples <- posterior_samples(x, "alpha")
+  q <- samples[, "alpha"]
+  if (!additive) {
```

Comment:
I find it easier to interpret q directly, i.e. 1 if no effect (and without calling it a percentage increase/decrease)